### PR TITLE
Feature: Add completed study page [LEI-255]

### PR DESCRIPTION
**Pull request by samanehsan**
If a user has already completed the study and tries to log in again, redirect them to a page that says they have already completed the study.

https://github.com/CenterForOpenScience/exp-addons/pull/168 needs to get merged when this goes in so that the feature flags still work

~~Note: Let's hold off on merging this until testing is complete so we can keep using the same test accounts~~

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -79,5 +79,9 @@ module.exports = function(environment) {
   // Whether to validate survey forms
   ENV.validate = false;
 
+  // Whether to redirect users who have already taken the study to an error page
+  // Set to false to test study multiple times with the same account
+  ENV.showStudyCompletedPage = false;
```
Comment (abought) [inline config/environment.js]: Is this a feature flag? If so, I encourage you to group them all together in a nested key, eg `ENV.featureFlags = {showStudyCompletedPage: false}`

File: app/templates/complete.hbs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<div class="exit-text text-center">
+  <p>{{t 'previouslogin.line1'}}</p>
```
Comment (abought) [inline app/templates/complete.hbs]: Feature question:

This text tells someone to contact their local ISP coordinator. Does the site provide any mechanism for contacting help? How does the user know who to contact, or how?

File: app/router.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ Router.map(function() {
     });
   });
   this.route('exit');
+  this.route('complete');
```
Comment (abought) [inline app/router.js]: Does this route make sense on its own? Is there an alternate arrangement that would make more sense to the user in the context of this application?

Eg `http://google.com/complete` vs `https://google.com/survey/complete`

Remember that once the product ships, it is much harder to change existing URLs.

**Review by abought — COMMENTED** [review]
No major blockers, but I just had a few small questions about this PR. We may want to think about routing (user-facing) and how we arrange feature flags (internal) before we go forward.

File: app/router.js (hunk 3 above)
Comment (samanehsan) [reply app/router.js]: Good question! Does it matter that a user is only supposed to see this route if they log in and have already completed the study? Meaning they will go from `https://google.com/ --> https://google.com/complete` or  `https://google.com/survey/complete` and not
`https://google.com/survey/ --> https://google.com/survey/complete`

File: app/templates/complete.hbs (hunk 2 above)
Comment (samanehsan) [reply app/templates/complete.hbs]: The email address of who to contact is in the consent form. In this case, they can't actually see that page again. But, my understanding is that the local ISP coordinator is the person who gave them their login info so they should know who to contact.

Comment (samanehsan) [reply app/templates/complete.hbs]: I'm sure we could add the local ISP coordinator's email address here, if that's what the client wants.

**Comment by abought** [comment]
> Good question! Does it matter that a user is only supposed to see this route if they log in and have already completed the study?

When you build a site, it's best to assume that you are stuck with the routes for the long term. The issue is not the path they took to get there- it's whether the route has meaning on its own.

For example, what happens if the site adds a second survey in the future, with a different completion page? Does this URL contain an implicit assumption that this is the only completion page they will ever interact with? Does the URL make as much sense, when read on its own, as the alternatives?

**Comment by samanehsan** [comment]
Thanks for the clarification @abought -- this is the only survey for the study right now but if they decide to do another run with an additional survey it's better to be flexible. I'll change it to `survey/complete`

File: app/router.js (hunk 3 above)
Comment (abought) [reply app/router.js]: Can you please clarify why the router defines the `complete` route twice?

(if this is legacy of a previous change, please glance over the PR to make sure nothing else is left over)

**Pull request merged**